### DISCUSSION TITLE
Server seen, but not working and no ping on explorer

### DISCUSSION THREAD

**Discussion by videobuff** [jamulussoftware/jamulus#1626] [Technical Support & Help]
To see if i can improve om latency. and performance, I installed a second jamukus server on a transip linux machine running Centos 8.
After previous expierence, Jamulus compiled fine and started with the same systemd scrpit as my other machine (the name has been chaged to The Sixties)flawlesly.

I have opened the firewall (CSF) on this server for the udp port and i can see the server on jamulus explorer. But i do not see the ping value, and it also does not appear in the client connect window.
I have scratched my head now for days what else i can do or test to make it usable.

Any help is much appreciated.

![Schermafbeelding 2021-05-04 om 18 55 30](https://user-images.githubusercontent.com/3132953/117042175-eb5e3f80-ad0b-11eb-8ffb-513f0750af5e.jpg)

**Comment by ghost** [comment]
I make sure you use two different port numbers on the servers.
eg. server1(Jamulus -s -p 22124)     server2(Jamulus -s -p 22125)
when they are on the same machine.

**Comment by videobuff** [comment]
David. They are on different servers.


Op di 4 mei 2021 om 19:47 schreef DavidSavinkoff ***@***.***>

> I make sure you use two different port numbers on the servers.
> eg. server1(Jamulus -s -p 22124) server2(Jamulus -s -p 22125)
> when they are on the same machine.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jamulussoftware/jamulus/discussions/1626#discussioncomment-695090>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAX44GNYLQTOQXPW7E4KIKTTMAXM5ANCNFSM44DDFBXQ>
> .
>
-- 
Verzonden via Gmail Mobile

**Reply by gene96817** [reply]
Are the servers sharing a firewall? If so, you need to configure the firewall port mapping.

**Reply by videobuff** [reply]
I am not sharing a server for the 2 Jamulus servers. They are on different linux servers in different locations (one is with transip, the other with liteserver). Both have their firewalls enabled and both have their udp ports open (22424)
The firewalls are different. One is CSF and one is Firewall-cmd. I understand that the clients can't reach my server, hence the lack of ping. But what other ports should I open up ?

**Reply by gene96817** [reply]
Can a client connect if you try to connect by the IP address?

**Reply by dcorson-ticino-com** [reply]
I tried connecting by IP,.... no response.

Could it be that your routing, forwarding, etc. is allowing outgoing packets, but not incoming ?
That would possibly point to the port forwarding not working.
In that case you could be registered on a Directory Server, requesting the server list, but never receiving it.

**Comment by dcorson-ticino-com** [comment]
Your server has registered with the directory server, but is not answering (or not receiving) the pings from the clients.
As thus it exists on the directory, but is not seen by any client.

Once the server is registered on a directory it works like this:
A client requests the server list from the directory.
Once the client has received the list of servers from the directory it pings all the servers on the list.
The servers are shown in the Connect list once the ping response has been received by the client.
If no response arrives the server is never shown in the Connect list.
Thus servers can be on the list of the directory that are not shown on a clients Connect list.

Why your server is not responding to the (UDP) pings of the clients, I do not have any ideas.

**Reply by softins** [reply]
> Your server has registered with the directory server, but is not answering (or not receiving) the pings from the clients.
> As thus it exists on the directory, but is not seen by any client.
> 
> Once the server is registered on a directory it works like this:
> A client requests the server list from the directory.

The directory sends the list to the client, _and_ the directory sends a request to every server in the list to send a message to the requesting client. When a server sends this message, that should cause a session to be established in the server's firewall that allows the client then to ping the server directly.

> Once the client has received the list of servers from the directory it pings all the servers on the list.
> The servers are shown in the Connect list once the ping response has been received by the client.
> If no response arrives the server is never shown in the Connect list.
> Thus servers can be on the list of the directory that are not shown on a clients Connect list.

Also, the Explorer back-end operates just like a client. If an entry shows an empty ping, version and OS, that means the Explorer back-end was unable to ping that particular server.

**Reply by gene96817** [reply]
It doesn't happen very often, occasionally an ISP could have a routing table error. The most difficult are small errors affecting a low traffic ISP-to-ISP hop.
